### PR TITLE
No longer lists all pacakges when no search term is provided

### PR DESCRIPTION
**Pull request by mbbaig**
Fix for #1462

### CODE REVIEW

**Comment by mbbaig** [comment]
Is the test needed any more?

**Comment by mbbaig** [comment]
Hi @sheerun thanks for responding.
could you review the code and let me know if the test I commented out is required.
Thanks :)

File: lib/commands/search.js [hunk 1]
```diff
@@ -14,12 +14,10 @@ function search(logger, name, config) {
...
-    // If no name was specified, list all packages
-    if (!name) {
-        return Q.nfcall(registryClient.list.bind(registryClient));
-    // Otherwise search it
-    } else {
+    if (name && !config.json) {
         return Q.nfcall(registryClient.search.bind(registryClient), name);
+    } else if (config.json) {
+        return Q.nfcall(registryClient.list.bind(registryClient));
     }
```
Comment (jisaacks) [inline lib/commands/search.js]: There used to be comments explaining the logic but now there are not. 

Why does `config.json` need to be present to search all?
Why does `config.json` need to be missing to search by name?
What happens when there is no `config.json` and no `name`?

File: templates/std/search-results.std [hunk 2]
```diff
@@ -6,5 +6,5 @@
 {{/.}}
 {{/condense}}
 
-{{else}}No results.
+{{else}}Please provide a valid search term. E.g. bower search angular
```
Comment (jisaacks) [inline templates/std/search-results.std]: What if a valid search term is provided by there are no results?

**Comment by jisaacks** [comment]
> if the test I commented out is required.

I think if the functionality to show all results when no keywords is removed then clearly the test is no longer needed. However it is probably wise to add tests for the new functionality. A test that ensures the correct response is given when there are no keywords and also when there are actually no results.

I am not a decision maker for this project, but that would be my 2¢

**Comment by mbbaig** [comment]
@jisaacks Valid points. I will look into adding and editing everything that you mentioned. Thanks

**Comment by BenMann** [comment]
Closing because it seems fixed already.
Here for details: https://github.com/bower/bower/issues/1462#issuecomment-190708468.

**Pull request closed without merging**